packages/react-charting/src/components/DonutChart/Pie/Pie.tsx

```diff
@@ -58,18 +69,70 @@ export class Pie extends React.Component<IPieProps, {}> {
     const focusData = this._pieForFocusRing(data);
     const piechart = pie(data);
     const translate = `translate(${this.props.width / 2}, ${this.props.height / 2})`;
+
+    this._getTotalValue();
+
     return (
       <g transform={translate}>
         {piechart.map((d: IArcData, i: number) => this.arcGenerator(d, i, focusData[i], this.props.href))}
+        {this._renderValueInsideDonut()}
       </g>
     );
   }
 
+  public componentDidUpdate(): void {
+    const classNames = getClassNames(getStyles, {
+      theme: this.props.theme!,
+    });
+
+    wrapTextInsideDonut(classNames.insideDonutString, this.props.innerRadius! * 2 - TEXT_PADDING);
+  }
+
   private _focusCallback = (data: IChartDataPoint, id: string, e: SVGPathElement): void => {
     this.props.onFocusCallback!(data, id, e);
   };
 
   private _hoverCallback(data: IChartDataPoint, e: React.MouseEvent<SVGPathElement>): void {
     this.props.hoverOnCallback!(data, e);
   }
+
+  private _renderValueInsideDonut = () => {
+    const { valueInsideDonut } = this.props;
```

Comment:
why move valueInsideDonut from arc to pie

Reply:
valueInsideDonut was rendered as many times as there were arcs due to which the cursor/focus moved to it multiple times

Reply:
This is good. Thanks